### DISCUSSION TITLE
Should warnings fail the migration?

### DISCUSSION THREAD

**Discussion by testerez** [prisma/prisma#11150] [Q&A] [answered]
Hello,
We just had a situation where a dev modified a migration file. This caused the migration to fail but the following execution of `prisma migrate deploy` just emitted a warning like this:
```
WARNING The following migrations have been modified since they were applied:
20211218190604_full_country_list
```
The problem is that this warning does not cause the migration to fail, so our builds passed happily even though the DB was not actually up to date. This caused the issue to remain unnoticed until we had production issues and realized our DB was not fully migrated.

I would expect the migration to fail in that situation so that we can't miss it and we fix the issue right away.

Does that make sense?

**Comment by samrathchadha** [comment, marked as answer]
This isn't technically a code breaking warning but if you've changed the SQL queries in your migration(which seems to be the case), you can always just run `npx prisma apply [query name]`

**Reply by testerez** [reply]
Thanks @yosamrath , we resolved the issue. My concern is more that I wish `prisma migrate deploy` would have failed so that our deployment was stopped and we discover the issue before our users.

**Reply by samrathchadha** [reply]
Ah ok, that's a very valid concern, you should definitely create an issue for it

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
